functions/lessonNotifications.js

```diff
@@ -0,0 +1,153 @@
+const { getFirestore } = require("firebase-admin/firestore");
+const { sendNotification, preparePayload } = require("./notification");
+
+
+const db = getFirestore();
+
+module.exports = {
+    // Notify student when the lesson is waiting for their review
+  notifyStudentLessonPendingReview: async (lesson) => {
+    const studentUid = lesson.studentUid;
+
+    try {
+      // Fetch the student profile
+      const studentDoc = await db.collection("profiles").where("uid", "==", studentUid).limit(1).get();
+
+      if (studentDoc.empty) {
+        console.error("Student profile not found.");
+        return;
+      }
+
+      const studentData = studentDoc.docs[0].data();
+      const studentPayload = preparePayload(
+        "Lesson Pending Review!",
+        `Your lesson with the tutor is now complete and is waiting for your review.`,
+        studentData.token
+      );
+
+      // Send notification
+      await sendNotification(studentPayload);
+    } catch (error) {
+      console.error("Error in notifyStudentLessonPendingReview:", error);
+    }
+  },
+
+    // Notify the student whenever an additional tutor offers to teach a lesson he requested
+  notifyStudentAsTutorOfferedToTeach: async (lesson) => {
+    const studentUid = lesson.studentUid;
+
+    try {
```

Comment:
Proper use of try-catch blocks ensures unexpected errors do not crash the system.